### PR TITLE
Binary pack script opt

### LINKED ISSUE #4003: Add service stop in binary/rpm/deb uninstall
This confused new users when they did this:

- remove everything on nebula and rpm -e
- install nebula
- The state of the cluster is the same as it was before, and it turned out the process are still the old ones.

We may add post script in remove phase to stop services.

### LINKED ISSUE #4009: Add order for script/service start all
To make storaged being started slightly slower than meta, to avoid storaged being waiting for a long time(really long) after the initial attempt of checking meta to be ready failed(due to meta is not yet up)

### PR DESCRIPTION
**Pull request by wey-gu**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 
close #4003 
close #4009 

#### Description:

- stop service before uninstalling packages
- try best to avoid storage initial start attempt failed due to meta is not yet ready(and the retry next attempt will be quite long, later)

## How do you solve it?

- Add pre uninstall script to stop service before uninstall
- Softly enforce waiting for port listening during service start with a best effort/timer

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [x] Manual Test(to be done later)

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....
- Binary package script optimized.

### CODE REVIEW

**Comment by wey-gu** [comment]
will manually test when I got bandwidth to, plz don't review now.

File: package/prerm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+install_dir=/usr/local/nebula
+daemons=(storaged metad graphd)
+for daemon in ${daemons[@]}
+do
+    if [[ ! -f $install_dir/scripts/nebula.service ]] ; then
+        $install_dir/scripts/nebula.service stop ${daemon}
```
Comment (CPWstatic) [inline package/prerm:8]: This operation can be dangerous in a production environment. I suggest that we do not do this implicitly.

File: scripts/nebula.service [hunk 2]
```diff
@@ -141,6 +141,8 @@ function start_daemon {
     local command="${executable} --flagfile ${config}"
     INFO "Starting ${daemon_name}..."
     eval ${command}
+    local port=$(get_port_from_config ${config})
+    wait_for_port_listening ${port} 20 || WARN "Timed out waiting for ${daemon_name} to finish starting, but it may still be running"
```
Comment (HarrisChu) [inline scripts/nebula.service:145]: prefer to implement this logic in nebula process.
and the behavior should be same with running in container.

Comment (HarrisChu) [reply scripts/nebula.service:145]: or just like readinessprobe in k8s, curl a http interface?

**Comment by wey-gu** [comment]
this will be revisited later when I have bandwidth to think about it.
Thanks for the reviews already ;)

**Comment by wey-gu** [comment]
closing it for now

**Pull request closed without merging**